### PR TITLE
[constants] remove deprecated fields, improve types

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-301

# How

This PR cleanups the Constants code for the SDK 44 release, and performs the following changes:

- Remove deprecated `Constants.deviceId`.
- Remove legacy `Constants.linkingUrl` alias.
- Add missing type declaration for `Constants.supportedExpoSdks` (but it's private/hidden field). 
- Add missing `null` to the `Constants.buildNumber` type.
- Fix duplicated `Manifest` type in manifest getter.

Also due to limited external solution for `installationId` replacement, it was decided to un-deprecate this field for a while.

# Test Plan

`yarn test` and `yarn lint` commands run in package directory doesn't yield any errors.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-constants/src/Constants.types.ts [hunk 1]
```diff
@@ -293,6 +293,7 @@ export interface NativeConstants {
    */
   systemFonts: string[];
   systemVersion?: number;
+  supportedExpoSdks?: string[];
```
Comment (brentvatne) [inline packages/expo-constants/src/Constants.types.ts:296]: let's mark this as private and be sure to leave it out of documentation

**Review by brentvatne — APPROVED** [review]

File: packages/expo-constants/src/Constants.types.ts (hunk 1 above)
Comment (Simek) [reply packages/expo-constants/src/Constants.types.ts:296]: So maybe adding this was a mistake? 😅

But I can see this array in managed workflows and in Snack, so for now I'm going to hide it and remove the changelog entry for this addition.

**Review by bbarthec — APPROVED** [review]

**Comment by PiotrSzlagura** [comment]
Could anyone tell how should we now get linkingUrl since this alias was removed? What is the original field?

In documentation it is still mentioned, eg. here: https://docs.expo.dev/versions/v45.0.0/sdk/auth-session/

**Comment by Simek** [comment]
There is no `linkingUrl` anymore. You can use `createURL()` method from `Linking` instead - https://docs.expo.dev/versions/latest/sdk/linking/#linkingcreateurlpath-namedparameters

Old deprecation note:
> Constants.linkingUrl has been renamed to Constants.linkingUri. Consider using the Linking API directly. Constants.linkingUrl will be removed in SDK 44.

**Comment by PiotrSzlagura** [comment]
@Simek Okay, I get it now. Would be good to update docs but I guess that's not in the scope of this issue. Thank you :)

**Comment by channeladam** [comment]
Just had the same problem... the inaccurate documentation caused me to waste a lot of time today...  https://docs.expo.dev/versions/v45.0.0/sdk/auth-session/

**Comment by Simek** [comment]
Thanks for pointing out the outdated places! I will update the docs soon, so we can avoid this confusion in the future.

**Pull request merged**